host/xtest/pkcs11_1000.c

```diff
@@ -8888,3 +8931,495 @@ static void xtest_pkcs11_test_1028(ADBG_Case_t *c)
+};
+
+#define CHUNK_SIZE 8
+
+static void xtest_pkcs11_test_1030(ADBG_Case_t *c)
+{
+	CK_RV rv = CKR_GENERAL_ERROR;
+	CK_SLOT_ID slot = 0;
+	CK_SESSION_HANDLE session = CK_INVALID_HANDLE;
+	CK_FLAGS session_flags = CKF_SERIAL_SESSION;
+	CK_OBJECT_HANDLE key_handle = CK_INVALID_HANDLE;
+	uint8_t in[512] = { 0 };
+	CK_ULONG in_size = 0;
+	uint8_t out[512] = { 0 };
+	CK_ULONG out_size = 0;
+	const struct aes_gcm_test *test = NULL;
+	size_t n = 0;
+	size_t proc_len = 0;
+	size_t total_len = 0;
+	uint8_t chunk[CHUNK_SIZE] = { 0 };
+	size_t chunk_len = 0;
+
+	rv = init_lib_and_find_token_slot(&slot);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		return;
+
+	rv = C_OpenSession(slot, session_flags, NULL, 0, &session);
+	if (!ADBG_EXPECT_CK_OK(c, rv))
+		goto err_close_lib;
+
+	for (n = 0; n < ARRAY_SIZE(cktest_aes_gcm_cases); n++) {
+
+		test = &cktest_aes_gcm_cases[n];
+		Do_ADBG_BeginSubCase(c, "AES-GCM case %zu)", n);
+
+		rv = C_CreateObject(session, test->attr_key, test->attr_count,
+				    &key_handle);
+		if (!ADBG_EXPECT_CK_OK(c, rv))
+			goto err;
+
+		/* Test 1 encrypt in 1 step */
+		if (test->ptx != NULL) {
+			rv = C_EncryptInit(session, test->mechanism,
+					   key_handle);
+			if (!ADBG_EXPECT_CK_OK(c, rv))
+				goto err_destr_obj;
+
+			/* Get output buffer length for update */
+			out_size = 0;
+			rv = C_EncryptUpdate(session, (void *)test->ptx,
+					     test->ptx_len, NULL, &out_size);
+			if (!ADBG_EXPECT_CK_OK(c, rv))
+				goto err_destr_obj;
+
+			/* Encrypt chunk */
+			total_len = 0;
+			memset(out, 0, out_size);
+			rv = C_EncryptUpdate(session, (void *)test->ptx,
+					     test->ptx_len, out, &out_size);
+			if (!ADBG_EXPECT_CK_OK(c, rv))
+				goto err_destr_obj;
+
+			total_len += out_size;
+
+			/* Finalize to get MAC */
+			rv = C_EncryptFinal(session, out + out_size, &out_size);
+			if (!ADBG_EXPECT_CK_OK(c, rv))
+				goto err_destr_obj;
+
+			total_len += out_size;
+
+			/* Check ciphertext and tag */
+			if (!ADBG_EXPECT_COMPARE_UNSIGNED(c,
+				test->ctx_len + test->tag_len, ==, total_len))
+				goto err_destr_obj;
+
+			if (!ADBG_EXPECT_BUFFER(c, test->tag,
+						 test->tag_len,
+						 out + test->ctx_len,
+						 test->tag_len))
+				goto err_destr_obj;
+
+			if (!ADBG_EXPECT_BUFFER(c, test->ctx,
+						 test->ctx_len,
+						 out, test->ctx_len))
+				goto err_destr_obj;
+		}
+
+		/* Test 2 decrypt in 1 step */
+		if (test->ptx != NULL) {
+			rv = C_DecryptInit(session, test->mechanism,
+					   key_handle);
+			if (!ADBG_EXPECT_CK_OK(c, rv))
+				goto err_destr_obj;
+
+			/* Concat ciphertext and tag */
+			memcpy(in, test->ctx, test->ctx_len);
+			memcpy(in + test->ctx_len, test->tag, test->tag_len);
+			in_size = test->ctx_len + test->tag_len;
+			total_len = 0;
+
+			/* Get output buffer length for update */
+			out_size = 0;
+			rv = C_DecryptUpdate(session, (void *)in, in_size,
+					     NULL, &out_size);
+			if (!ADBG_EXPECT_CK_OK(c, rv))
+				goto err_destr_obj;
```

Comment:
There is not output data returned until GCM final stage (and if the tag is attests the payload).
So no query of the output size here. You can even test `ADBG_EXPECT_COMPARE_UNSIGNED(c, out_size, ==, 0)`.